lib/common/http1client.c

```diff
@@ -553,13 +556,21 @@ static void on_head(h2o_socket_t *sock, const char *err)
         on_head.pipe_reader = &client->pipe_reader;
 #endif
 
+    client->steal_bytes = (size_t)rlen;
```

Comment:
Rather than adding `steal_bytes` as a member of `h2o_http1client_t`, how about passing `rlen` as an argument to `call_on_head`? By doing so, the code becomes clearer regarding when the value can be non-zero.

Reply:
> how about passing rlen as an argument to call_on_head? By doing so, the code becomes clearer regarding when the value can be non-zero.

Addressed in https://github.com/h2o/h2o/pull/3405/commits/494436bf0dc27f3743fb7d48a1873ed44fbfaf55, but

> Rather than adding steal_bytes as a member of h2o_http1client_t

We still need to store that rlen value as a member of client, because we should provide it in get_conn_properties callback.